src/test/regress/sql/offload_entry_to_qe.sql

```diff
@@ -0,0 +1,119 @@
+set optimizer = off; -- orca is currently unsupported
+set enable_offload_entry_to_qe = on;
+
+create temp table tst(x int, y int);
+create index tst_y_idx on tst(y);
+insert into tst values (1, 1), (1, 1), (2, 1), (10, 10);
+analyze tst;
+
+-- accept Aggregate, Finalize Aggregate, Limit + Finalize Aggregate, WindowAgg, Sort + Unique
+explain (costs off, locus) select count(x) from tst where x > 1;
+explain (costs off, locus) select count(x) from tst;
+explain (costs off, locus) select count(x) from tst limit 1;
+explain (costs off, locus) select count(x) over () from tst;
+explain (costs off, locus) select count(x) over () from tst group by x limit 1;
+explain (costs off, locus) select distinct min(x), max(x) from tst;
+-- compare results with GUC set to false
+set enable_offload_entry_to_qe = off;
+select count(x) from tst where x > 1;
+select count(x) from tst;
+select count(x) from tst limit 1;
+select count(x) over () from tst;
+select count(x) over () from tst group by x limit 1;
+select distinct min(x), max(x) from tst;
+set enable_offload_entry_to_qe = on;
+select count(x) from tst where x > 1;
+select count(x) from tst;
+select count(x) from tst limit 1;
+select count(x) over () from tst;
+select count(x) over () from tst group by x limit 1;
+select distinct min(x), max(x) from tst;
+
+-- accept Merge Join, Nested Loop and Hash Join
+set enable_seqscan = off;
+set enable_nestloop = off;
+set enable_mergejoin = on;
+explain (costs off, locus) select * from tst t1, tst t2 where t1.y = t2.y order by t1.y;
+set enable_offload_entry_to_qe = off;
+select * from tst t1, tst t2 where t1.y = t2.y order by t1.y;
+set enable_offload_entry_to_qe = on;
+select * from tst t1, tst t2 where t1.y = t2.y order by t1.y;
+reset enable_mergejoin;
+reset enable_nestloop;
+reset enable_seqscan;
+explain (costs off, locus) select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on true;
+explain (costs off, locus) select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on t1.x = v.x;
+set enable_offload_entry_to_qe = off;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on true;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on t1.x = v.x;
+set enable_offload_entry_to_qe = on;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on true;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on t1.x = v.x;
```

Comment:
add a cte test, I guess .

Reply:
I tried CTE but I found it a bit tricky. for CTE, `root->numMotions` always returns `0` so that offloading are always disabled. also CTE has *hidden* `SubqueryScan` that's currently unsupported. I'll try to come up with a newer version of code that supports CTE.

Reply:
cte: root->numMotions = 0 ， is this a bug ?

Reply:
I don't think so. I think maybe PG handles each `WITH` table independently. the tables references in a `WITH` clause even doesn't exist in the outermost query's range table list.